src/app/container/paramfiles/paramfiles.service.ts

```diff
@@ -55,8 +56,69 @@ export class ParamfilesService {
     return descriptorsWithParamfiles;
   }
 
+  /**
+   * Gets the descriptor types (CWL/WDL/NFL) that are valid and have valid test parameter files
+   * @param version the current selected version of the workflow or tool
+   * @returns an array that may contain 'CWL' or 'WDL' or 'NFL'
+   */
+    getValidDescriptors(version: (WorkflowVersion|Tag)) {
+      if (version) {
+        const descriptorTypes: Array<ToolDescriptor.TypeEnum> = [];
+        if (version.validations) {
+          if (this.checkValidFileType(version, SourceFile.TypeEnum.CWLTESTJSON, SourceFile.TypeEnum.DOCKSTORECWL)) {
+            descriptorTypes.push(ToolDescriptor.TypeEnum.CWL);
+          }
+
+          if (this.checkValidFileType(version, SourceFile.TypeEnum.WDLTESTJSON, SourceFile.TypeEnum.DOCKSTOREWDL)) {
+            descriptorTypes.push(ToolDescriptor.TypeEnum.WDL);
+          }
+
+          if (this.checkValidFileType(version, SourceFile.TypeEnum.NEXTFLOWTESTPARAMS, SourceFile.TypeEnum.NEXTFLOWCONFIG)) {
+            descriptorTypes.push(ToolDescriptor.TypeEnum.NFL);
+          }
+        }
+        return descriptorTypes;
+      }
+    }
+
+    /**
+     * A helper method that checks for a given version and file type,
+     * if the file type has a valid validation and at least one file is present.
+     * The related language must have a valid descriptor too.
+     * All these must be true to show the corresponding language in the dropdown
+     * @param version Version of interest
+     * @param fileType Ex. SourceFile.TypeEnum.CWLTESTJSON
+     * @param descriptorType Ex. SourceFile.TypeEnum.DOCKSTORECWL
+     */
+    checkValidFileType(version: (WorkflowVersion|Tag), fileType: SourceFile.TypeEnum, descriptorType: SourceFile.TypeEnum) {
+      // Check that the language has a valid descriptor
+      const descriptorValidation = version.validations.find((validation: Validation) => {
+        return validation.type === descriptorType;
+      });
+
+      if (!(descriptorValidation && descriptorValidation.valid)) {
+        return false;
+      }
+
+      // Check that at least one file is valid
+      const validationObject = version.validations.find((validation: Validation) => {
+        return validation.type === fileType;
+      });
+
+      if (!(validationObject && validationObject.valid)) {
+        return false;
+      }
+
+      // Check that at least one file is present
+      const langaugeFile = version.sourceFiles.find((sourceFile) => {
```

Comment:
You could optionally use `some` instead of `find`, as `some` already returns a boolean.